05. 효율성(투포인터 알고리즘, 슬라이디윈도우, 해쉬)/07. 아나그램/김보경/index.js

```diff
@@ -1,9 +1,43 @@
 function solution(str1, str2) {
-  let answer = 'YES';
+  let answer = "YES";
+  let str1Count = {};
+  let str2Count = {};
 
-  return answer;
+  if (str1.length !== str2.length) return "NO";
+
+  [...str1].forEach((el) => {
+    if (el in str1Count) {
+      str1Count[el]++;
+    } else {
+      str1Count[el] = 1;
+    }
+  });
+  [...str2].forEach((el) => {
+    if (el in str2Count) {
+      str2Count[el]++;
+    } else {
+      str2Count[el] = 1;
+    }
+  });
+
+  let str1Result = [];
+  let str2Result = [];
+
+  for (let key in str1Count) {
+    let count = key + str1Count[key];
+    str1Result.push(count);
+  }
+  for (let key in str2Count) {
+    let count = key + str2Count[key];
+    str2Result.push(count);
+  }
+
+  str1Result.sort().join("");
+  str2Result.sort().join("");
+
+  return str1Result.includes(str2Result) ? answer : "NO";
```

Comment:
`Array.prototype.join`은 새로운 문자열을 만들어서 반환하는데 그걸 활용하지 않아서 오답 처리가 되는 거 같아요!

```suggestion
  str1Result = str1Result.sort().join("");
  str2Result = str2Result.sort().join("");

  return str1Result.includes(str2Result) ? answer : "NO";
```